### PR TITLE
Removed the CriticalAddOn toleration as it is not used

### PR DESCRIPTION
**Pull request by thorfour**
Removing the CriticalAddonsOnly toleration as it isn't being used.

### CODE REVIEW

File: modules/net/canal/resources/manifests/kube-calico.yaml [hunk 1]
```diff
@@ -126,8 +126,7 @@ spec:
       labels:
         k8s-app: kube-calico
       annotations:
-        # This, along with the CriticalAddonsOnly toleration below,
-        # marks the pod as a critical add-on, ensuring it gets
+        # This marks the pod as a critical add-on, ensuring it gets
         # priority scheduling and that its resources are reserved
         # if it ever gets evicted.
         scheduler.alpha.kubernetes.io/critical-pod: ''
```
Comment (squat) [inline modules/net/canal/resources/manifests/kube-calico.yaml]: @thorfour if the pod has the critical-pod annotation then it is required to have the criticaladdonsonly toleration to be guaranteed space. Without the toleration:
1. the node will get the CriticalAddonsOnly taint
2. the scheduler will try to schedule the kube-calico pod but since it doesn't tolerate the correct taint, it will not be schedulable
3. so the rescheduler will keep evicting pods from the nodes until the kube-calico pod can be scheduled (which it will never be)
4. go to 2

TL;DR: if we are removing criticaladdonsonly tolerations then we must remove this annotation as well.

More info: https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/

File: modules/tectonic/resources/manifests/heapster/deployment.yaml [hunk 2]
```diff
@@ -58,6 +58,3 @@ spec:
             requests:
               cpu: 50m
               memory: 90Mi
-      tolerations:
```
Comment (squat) [inline modules/tectonic/resources/manifests/heapster/deployment.yaml]: we need to remove the annotation in this manifest as well.

File: modules/bootkube/resources/manifests/kube-dns.yaml [hunk 3]
```diff
@@ -162,8 +162,6 @@ spec:
             cpu: 10m
       dnsPolicy: Default  # Don't use cluster DNS.
       tolerations:
-      - key: "CriticalAddonsOnly"
```
Comment (squat) [inline modules/bootkube/resources/manifests/kube-dns.yaml]: we need to remove the annotation in this manifest as well.

File: modules/net/canal/resources/manifests/kube-calico.yaml (hunk 1 above)
Comment (thorfour) [reply modules/net/canal/resources/manifests/kube-calico.yaml]: Good to know, thanks for that information. I've removed the remaining annotations.

**Comment by yifan-gu** [comment]
We want to get this in for 179 right? Cuz I saw all other critical pod annotations are removed in 179.

**Comment by squat** [comment]
ok to test

**Comment by yifan-gu** [comment]
PR merged, I also rebased 1.7.9 branch to master to pick up this one.
